### ISSUE TITLE
Nested collections not being displayed consistently

### ISSUE THREAD

**Issue by t3mujin** [kaiiiz/obsidian-raindrop-highlights-plugin#36]
I have some nested collections in my raindrop, and when I select the Manage option the list showing in the modal window is displayed inconsistently: some collections are shown with the full hierarchical name while others just with the collection name, this leads to collections with the same parent being pulled entirely differently, because this also as impact on the import.

The Trekking collections in Raindrop are inside Clippings/Travel, but here are shown separately

![image](https://user-images.githubusercontent.com/331767/216029202-737e560c-2bd0-406b-b720-cc32466ea0ad.png)

**Comment by kaiiiz** [comment]
Thanks for reporting the bug. However, I'm currently not a Raindrop PRO user. Can you provide the API responses for the following two endpoints: `https://api.raindrop.io/rest/v1/collections` and `https://api.raindrop.io/rest/v1/collections/childrens`?

Example using curl:

```
curl -H "Authorization: Bearer <replace this to your token>" https://api.raindrop.io/rest/v1/collections
```

```
curl -H "Authorization: Bearer <replace this to your token>" https://api.raindrop.io/rest/v1/collections/childrens
```

Thanks.

**Comment by gwillcox-r7** [comment]
Don't know if this will help @kaiiiz but here is my response. `Go` and `Ruby` are nested inside the `Programming Languages` collection but they aren't showing up here.

```
> curl -H "Authorization: Bearer *redacted*" https://api.raindrop.io/rest/v1/collections
{"result":true,"items":[{"title":"Obsidian Learning","description":"","public":false,"view":"grid","count":31,"cover":[],"expanded":false,"_id":25691070,"user":{"$ref":"users","$id":981098,"$db":""},"creatorRef":{"_id":981098,"name":"tekwizz123","email":""},"lastAction":"2022-10-16T01:34:58.857Z","created":"2022-07-03T21:54:33.585Z","lastUpdate":"2022-10-16T01:34:58.858Z","sort":25691070,"slug":"obsidian-learning","access":{"for":981098,"level":4,"root":false,"draggable":true},"author":true},{"title":"Archive","description":"","public":false,"view":"list","count":79,"cover":["https://up.raindrop.io/collection/thumbs/256/901/16/5d13a4304e76fcdf386389e9e5a47e95.png"],"expanded":false,"_id":25690116,"user":{"$ref":"users","$id":981098,"$db":""},"creatorRef":{"_id":981098,"name":"tekwizz123","email":""},"lastAction":"2022-11-29T21:25:41.160Z","created":"2022-07-03T21:11:41.767Z","lastUpdate":"2022-11-29T21:25:41.160Z","sort":25690116,"slug":"archive","color":"#1e76cf","access":{"for":981098,"level":4,"root":false,"draggable":true},"author":true},{"title":"Self Improvement","description":"","public":false,"view":"grid","count":2,"cover":[],"expanded":false,"_id":25851968,"user":{"$ref":"users","$id":981098,"$db":""},"creatorRef":{"_id":981098,"name":"tekwizz123","email":""},"lastAction":"2022-09-24T02:44:05.093Z","created":"2022-07-13T16:20:14.291Z","lastUpdate":"2022-09-24T02:44:05.093Z","sort":14,"slug":"self-improvement","access":{"for":981098,"level":4,"root":false,"draggable":true},"author":true},{"title":"Training for Developer Role","description":"","public":false,"view":"grid","count":7,"cover":[],"expanded":false,"_id":25851931,"sort":5,"user":{"$ref":"users","$id":981098,"$db":""},"creatorRef":{"_id":981098,"name":"tekwizz123","email":""},"lastAction":"2022-09-23T04:22:20.131Z","created":"2022-07-13T16:15:57.380Z","lastUpdate":"2022-09-23T04:22:20.131Z","slug":"training-for-developer-role","access":{"for":981098,"level":4,"root":false,"draggable":true},"author":true},{"title":"Research To Read Through","description":"","public":false,"view":"list","count":137,"cover":[],"expanded":false,"_id":25851970,"user":{"$ref":"users","$id":981098,"$db":""},"creatorRef":{"_id":981098,"name":"tekwizz123","email":""},"lastAction":"2022-10-24T14:22:38.983Z","created":"2022-07-13T16:20:15.608Z","lastUpdate":"2022-10-24T14:22:38.983Z","sort":2,"slug":"research-to-read-through","access":{"for":981098,"level":4,"root":false,"draggable":true},"author":true},{"title":"Programming Languages","description":"","public":false,"view":"grid","count":2,"cover":[],"expanded":true,"_id":25869534,"sort":2,"user":{"$ref":"users","$id":981098,"$db":""},"creatorRef":{"_id":981098,"name":"tekwizz123","email":""},"lastAction":"2022-09-23T04:22:20.131Z","created":"2022-07-14T15:47:46.394Z","lastUpdate":"2022-09-26T04:20:06.488Z","slug":"programming-languages","access":{"for":981098,"level":4,"root":false,"draggable":true},"author":true},{"title":"Metasploit Evasion","description":"Metasploit Evasion Writeups","public":true,"view":"grid","count":82,"cover":[],"expanded":false,"_id":25826921,"user":{"$ref":"users","$id":981098,"$db":""},"creatorRef":{"_id":981098,"name":"tekwizz123","email":""},"lastAction":"2022-11-29T06:30:33.849Z","created":"2022-07-12T14:52:17.429Z","lastUpdate":"2022-11-29T06:30:33.849Z","sort":1,"slug":"metasploit-evasion","access":{"for":981098,"level":4,"root":false,"draggable":true},"author":true},{"title":"Maintaining OSS","description":"","public":false,"view":"grid","count":8,"cover":[],"expanded":false,"_id":27246991,"user":{"$ref":"users","$id":981098,"$db":""},"creatorRef":{"_id":981098,"name":"tekwizz123","email":""},"lastAction":"2022-12-19T16:57:31.784Z","created":"2022-09-01T14:49:24.456Z","lastUpdate":"2022-12-19T16:57:31.784Z","sort":1,"slug":"maintaining-oss","access":{"for":981098,"level":4,"root":false,"draggable":true},"author":true},{"title":"Windows Internals","description":"","public":false,"view":"list","count":65,"cover":[],"expanded":false,"_id":25909719,"sort":0,"user":{"$ref":"users","$id":981098,"$db":""},"creatorRef":{"_id":981098,"name":"tekwizz123","email":""},"lastAction":"2022-11-29T21:23:59.645Z","created":"2022-07-16T15:06:44.722Z","lastUpdate":"2022-11-29T21:23:59.645Z","slug":"windows-internals","access":{"for":981098,"level":4,"root":false,"draggable":true},"author":true},{"title":"Useful Administrative Tips","description":"","public":false,"view":"grid","count":12,"cover":[],"expanded":false,"_id":25910423,"sort":0,"user":{"$ref":"users","$id":981098,"$db":""},"creatorRef":{"_id":981098,"name":"tekwizz123","email":""},"lastAction":"2022-07-17T04:06:53.858Z","created":"2022-07-16T16:48:32.062Z","lastUpdate":"2022-09-22T14:13:42.359Z","slug":"useful-administrative-tips","access":{"for":981098,"level":4,"root":false,"draggable":true},"author":true}]}%    
```

**Comment by kaiiiz** [comment]
@gwillcox-r7 thanks for your help. The code seems to work on nested two layers of collections. Can you provide another API response for more than two layers? Thanks.

**Comment by BearCarpenter** [comment]
I have the same problem and when I click 'sync' then I see the message:
Sync Raindrop collection X/Y failed. Cannot read properties of null (reading 'path')
I have that structure in rain:
X/Y/Z

Y and Z is folder.
Y and Z has bookmarks too.
Y has folder and bookmarks.
Z has only bookmarks.

**Comment by kaiiiz** [comment]
The bug is caused by the irregular order of collections in the `children` endpoint. This should be fixed in v0.0.16.
